R/use-bbi.R

```diff
@@ -225,89 +219,52 @@ build_bbi_install_path <- function() {
-  } else {
-    bbi_commands <- nx_install_commands(.path, .bbi_url)
+  tmpdir <- tempfile()
+  dir.create(tmpdir)
+  orig_wd <- setwd(dir = tmpdir)
+  on.exit(setwd(orig_wd), add = TRUE)
+  on.exit(unlink(tmpdir, recursive = TRUE), add = TRUE)
+
+  download_target <- file.path(tmpdir, "bbi.tar.gz")
+  be_verbose <- isTRUE(getOption("bbr.verbose"))
+
+  rc <- download_with_retry(
+    .bbi_url, destfile = download_target, mode = "wb",
+    quiet = isFALSE(be_verbose))
+  if (rc != 0) {
+    stop(glue("Non-zero exit ({rc}) for download of {.bbi_url}"))
   }
-  return(bbi_commands)
-}
 
+  utils::untar(tarfile = download_target, verbose = be_verbose)
 
-#' Private helper function for building commands to install bbi on Linux or Unix
-#' @inheritParams build_bbi_install_commands
-#' @importFrom stringr str_replace
-#' @keywords internal
-nx_install_commands <- function(.path, .bbi_url) {
-  # extract dir name that tar will unzip to
-  tar_basename <- stringr::str_replace(basename(.bbi_url), "\\..+$", "")
-
-  # construct commands
-  return(c(
-    glue('wget {.bbi_url} -O /tmp/bbi.tar.gz  --timeout=15 --tries=2'),
-    glue('rm -rf /tmp/{tar_basename}'),
-    'tar -xzf /tmp/bbi.tar.gz -C /tmp/',
-    glue('mv /tmp/{tar_basename}/bbi {.path}'),
-    glue('chmod +x {.path}')
-  ))
-}
+  # The extracted directory should have only one directory...
+  d_extracted <-  list.dirs(tmpdir, recursive = FALSE)
+  checkmate::assert_string(d_extracted, min.chars = 1)
+  # ... with one executable inside.
+  exe <- list.files(d_extracted, full.names = TRUE)
+  checkmate::assert_string(exe, min.chars = 1)
 
-#' Private helper function for building commands to install bbi on Windows
-#'
-#' THIS HAS NOT BEEN TESTED!!! Honestly, I have not idea if the file.path()
-#' will work or not, but passing backslashes around in strings in R is
-#' generally very shaky ground.
-#' @inheritParams build_bbi_install_commands
-#' @importFrom stringr str_replace
-#' @keywords internal
-windows_install_commands <- function(.path, .bbi_url) {
-  # check for wget, these people claim it's possible:
-  # https://builtvisible.com/download-your-website-with-wget/
-  if (Sys.which("wget") == "") {
-    stop("Must install `wget` to install with use_bbi() on Windows", call. = FALSE)
-  }
-  # same for tar
-  # https://wiki.haskell.org/How_to_unpack_a_tar_file_in_Windows
-  if (Sys.which("tar") == "") {
-    stop("Must install `tar` to install with use_bbi() on Windows", call. = FALSE)
+  Sys.chmod(exe, mode = "0755", use_umask = FALSE)
+  if (be_verbose) {
+    message(glue("Copying bbi to '{.path}'"))
   }
+  file.copy(exe, .path, overwrite = TRUE)
 
-  # extract dir name that tar will unzip to
-  tar_basename <- stringr::str_replace(basename(.bbi_url), "\\..+$", "")
-
-  # construct commands
-  # WE USE ONLY THE install_dir, because I can't get a straight answer about Windows temp dirs
-  temp_dir <- file.path(dirname(.path), "TEMP")
```

Comment:
With a quick search, I didn't find anything about `tempfile` not working on Windows.  @dpastoor, if you have a moment, can you check whether `tempfile` seems to behave sensibly on Windows for you (not in the context of this PR, just a plain call to `tempfile`)?